clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -878,6 +878,7 @@ public void assign(Collection<TopicPartition> partitions) {
                 remaining = timeout - elapsed;
             } while (remaining > 0);
 
+            client.quickPoll(false, false);
```

Comment:
Can you explain why this case is different than the one above? Why do we not fail unsent requests?